src/network/networkd-link.c

```diff
@@ -242,6 +243,15 @@ void link_update_operstate(Link *link, bool also_update_master) {
         else
                 operstate = LINK_OPERSTATE_ENSLAVED;
 
+        /* Only determine online state for managed links with RequiredForOnline=yes */
+        if (!link->network || !link->network->required_for_online)
+                online_state = _LINK_ONLINE_STATE_INVALID;
```

Comment:
Hmm, I think we should also set online state for unmanaged links.
How about the following?
```
if (operstate >= link->network ? link->network->required_operstate_for_online.min : LINK_OPERSTATE_RANGE_DEFAULT.min &&
    operstate <= link->network ? link->network->required_operstate_for_online.max : LINK_OPERSTATE_RANGE_DEFAULT.max)
        online_state = LINK_ONLINE_STATE_ONLINE;
else
        online_state = LINK_ONLINE_STATE_OFFLINE;
```

Reply:
Hmm, what value would that bring? For unmanaged links, `LINK_OPERSTATE_RANGE_DEFAULT` is only a best guess on the part of networkd, since it does not know how the link should be configured. But it can be potentially misleading too. Why not be transparent about it?

For managed links where `RequiredForOffline=no`, I would argue the same way. By setting `=no`, the admin has no way to configure the actual `required_operstate_for_online`, so it will be a best guess.

I can of course do what you suggest - I don't think it's wrong per se - but I invite you to reconsider first.